thirtyinch/src/main/java/net/grandcentrix/thirtyinch/distinctuntilchanged/DistinctUntilChangedInvocationHandler.java

```diff
@@ -28,15 +29,15 @@
 
     private static final String TAG = DistinctUntilChangedInvocationHandler.class.getSimpleName();
 
-    private HashMap<String, Integer> mLatestMethodCalls = new HashMap<>();
+    private HashMap<String, WeakReference<Object[]>> mLatestMethodCalls = new HashMap<>();
```

Comment:
I do not see how `WeakHashMap` is benefit here, since it allows keys to be garbage collected. Since we put method names as keys and method arguments as values and I doubt our method name will be gc'ed.